### DISCUSSION TITLE
Unable to add files on macOS 12 INTEL

### DISCUSSION THREAD

**Discussion by Fred-IV** [yermak/AudioBookConverter#581] [MacOS]
Intel Mac Pro DP, 128GB RAM

Unable to add or open files. I tried Drag&Drop, as well as open/add files.
Maybe it doesn't work at all for what I am trying to do, but I get no error messages, just nothing happens.

I am trying to open existing m4b files I created to fix them.
If this app cannot work with m4? files, please let me know.
I also tried the conversion menu, and nothing...

My problem; I converted many of my CD AudioBook collections to m4b, maybe not using the apps right.
I did it with AudioBook Binder and/or AudioBook Builder. I was trying both.

The issue; As I am going through my drives, and doing housekeeping, I have discovered that many no longer work with QuickLook. It just freezes.
Some do work, and I can rename them from m4b to m4a and they still work.

AudioBook Builder and Binder have a major drawback, neither can edit existing files, so that's how I found this program. I am looking for an app to possibly fix my files, and not have to start from scratch. My chapters and cover art are already there, and starting over would suck.

I found another program called Switch, but it's really bad. It bloats up the file size 50-150%, and there are no settings to work with.

If this app can't do what I need, anyone know of one that will?

QuickLook won't play the damaged ones, but QuickTime will, so I don't think it's a DRM thing. I didn't save any of my projects with DRM.
I think one of the AudioBook apps is bad, and I will try to figure out which later, right now, I just want to fix my files.

I tried to open the .m4b files with a hex editor to see if there was a signature in the file as to which created it, but there is not. Neither on the working or the broken files.

I can get info on the files in QT, and I can see that the broken ones have a rogue video track, the working ones do not, so I am sure one of the builder apps is bad.

I need an app that can edit and delete the video track, or re-save it with options or conversion.
Simply renaming from m4b to m4a doesn't get around this issue. It only works for the files that already work with no video track.

When loading the broken files with QT I do get an error, but it can be played.
Screenshots:
1) error on damaged file at QT load.
2) QT info showing the rogue video track
3) QT info of the file that works

So, if anyone knows how to make AudioBookConverter work on these files, or of another app that can help, please let me know. I hate to start from scratch.
It just needs to be able to retain my album art and chapters.

Thanks!

![CleanShot 2023-06-10 at 05 49 00](https://github.com/yermak/AudioBookConverter/assets/66499758/91c05fba-8a94-4963-811b-330c93cba909)
![CleanShot 2023-06-10 at 05 51 43](https://github.com/yermak/AudioBookConverter/assets/66499758/d4d83a5a-2623-4409-98b6-1a17c89eb15c)
![CleanShot 2023-06-10 at 05 52 34](https://github.com/yermak/AudioBookConverter/assets/66499758/bfc2e9b1-bad4-42f6-93a2-84ade3e5555a)

**Comment by Fred-IV** [comment]
Actually, it's not just m4? files that won't work.
This app doesn't seem to work at all.
I have tried to add mp3 files to see what it can do, and same issue. No response.
I f Drag&Drop the folder, or mp3 files, nothing happens.
If I try open, I don't get a file open dialog box.
Maybe this thing doesn't work at all on macOS 12, or there's some other bug.
It seems that nothing works.
The only menu item that responded was changing my settings to dark mode.
It's not locked up, it just doesn't do anything.

**Comment by Fred-IV** [comment]
I found out which corrupts files.
AudioBook Builder is the problem. It doesn't work at all.
AudioBook Binder works great, but they just went out of business.
Their website says so long...
It's still on the app store, but who knows for how long.

**Comment by yermak** [comment]
Hi @Fred-IV ,
This program is not very stable under new MacOS, it seems they changed permissions again, i can't keep up with their changes using this program as it has long legacy roots. I'm rebuilding this program completely in native maner for MacOS, can promise exact release date - but it will be soon.

I think it stopped working as they program can't access any files w/o explicit permission, while it still may work under older MacOS - I don't think investing time into workaround for MacOS (with not guaranteed result) changes is wise, comparing to building new program which will work better.

Did you solve you you problem?

This program can open existing m4b and re-package ( and/or re-encode) them into m4b/mp3.
If you have windows PC maybe you can try windows version for now.

**Comment by dchuk** [comment]
> I'm rebuilding this program completely in native maner for MacOS, can promise exact release date - but it will be soon.

Is there anywhere we can follow along with your work on this?

**Comment by taw123** [comment]
At the risk of hijacking this thread (bad forum non my part), I'd like to add that I TOO can not add files to v6.1.2 using Intel. That said I am running this on Catalina so don't think I should be effected by the NEWEST permission problems discussed above.

Specific config: 
- OS: Mac OS 10.15.7 (19H1824)
- Audio-Converter 6.1.2
- Files attempted: mp3


I have tried as the OP discussed:
1. Drag to the finder (OS shows the + sign but nothing added in the window
2. Clicked File MENU in the WINDOW of the application: neither add folder or file does anything beyond allowing for selection in UI
3. Clicked add in the Queue tab this time clicking on either add file or folder DOSEN'T clear the menu selection (leaving File or Folder selected) and of course command NOT executed.

Nothing particularly interesting in the console log either (was a dev myself for years at a certain fruit company if that helps and you would like further info on app status/enviorment).

Is there a PRIOR version of the application that might have better results (given the vintage nature of this environment)? Or is there a more general issue with file actions/permissions? The application is running from the application directory and has had it's quarantine status cleared.    

Thanks! And again apologies if I am hijacking however I did think the symptoms if not the complete environments were similar enough to warrant adding here in hops of potentially resolving both your cases from a potential root cause...

**Comment by taw123** [comment]
Followup:

Ever the former Engineer, I attempted to regress the problem by trying an earlier release (not all testing done with releases from GitHub, as this made for the easiest testing approach, would be willing to purchase via Steam or direct payment if desired but given current lack of success that seem pointless at this juncture).

Continuing...  I retried same set up above Mac OS 10.15.7 (19H1824) with mp3 files but with the last version 5 release (5.6.2) again pulled from the DMG. THIS time I WAS able to add files to the UI (via menus AND drag/drop). I SPECIFICALLY declined the upgrade dialog for testing. UNFORTUNATELY when I attempted to start the audiobook conversion after bing prompted for a target destination/file name ALL content in the application vanished leaving the application the empty (no files, no name/description/etc) just it shows when launched. Again nothing particularly interesting in the log though there is a good deal of data logged if you enable debug & info messages.

One last lark I tried adding both current release and v5.6.2 MANUALLY to Full Disk access privileges to see it this might provide further insights. This had NO EFFECT on the current release HOWEVER when re-testing 5.6.2 the app now hangs when one attempts to start the audiobook creation rather then just clearing out all the info as it did in the paragraph above. Again no useful console data in terms of errors but can provide logs with info&debug on as there is much more data you are pushing there if it will assist you.

At this point I don't see much point in trying random releases without guidance so I'll wait for an update on the bug by the dev and try another application while waiting. 

Please let me know if there is anything further I can provide to assist in diagnosis. I know you are working on a re-write but I am assuming that my config IS/WAS a supported config at some point and SHOULD work.... :)

**Comment by appleplants** [comment]
i had the same issue with the mac build, i tried porting the windows exe to mac using porting kit, and it works fine with most audio files now. i used the ws11wineCX64bit22.1.1 wine engine and the latest 6.1.2 audiobookconverter exe release. so if anyone's pressed, it could be a fix.

**Comment by ZolaLa9** [comment]
> I'm rebuilding this program completely in native maner for MacOS, can promise exact release date - but it will be soon.

Hello @yermak — are there still plans to rebuild and offer this program for macOS? thank you

**Comment by yermak** [comment]
Duplicates #580

**Comment by yermak** [comment]
Beta for Mac OS X is available
https://github.com/yermak/AudioBookConverter/issues/580#issuecomment-2268738667
